### ISSUE TITLE
Stop cursor blinking after idling for x minutes

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#4628]
Context: https://github.com/xtermjs/xterm.js/issues/4082#issuecomment-1660296774

> To solve the power problem, maybe we should just detect when the terminal is focused and idle as that's the main case I'm concerned about. Something like a setTimeout(2 * 60 * 1000) which clears on focus out and resets on any input. Let's tackle that separately to https://github.com/xtermjs/xterm.js/pull/4568 though

**Comment by Tyriar** [comment]
The `IdleDetector` API looks great for this, but it requires permission https://developer.chrome.com/articles/idle-detection/

**Comment by Tyriar** [comment]
It also can only be used on user activation

**Comment by jerch** [comment]
Yeah the IdleDetector API sounds too invasive, ppl would have many reasons not to allow that.

Would a heuristic respecting things like "not focused", "no mouse move for x seconds" or "no user input/write attempt for y seconds" do? This still would fail on ppl just sitting in front of the computer staring at the display, but imho the IdleDetector API would have the same issue.
